docs/spec.openapi.yaml

```diff
@@ -0,0 +1,258 @@
+        '404':
+          description: Not found. No directory with the given parent_uuid was found.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error_response'
+        '500':
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error_response'
+
+    post:
+      tags: [ 'File' ]
+      description: Save the metadata for a new file or directory.
+      parameters:
+        - in: path
+          name: user_uuid
+          schema:
+            type: string
+            example: '658b4e63-b5ac-46a7-ac43-efb6a1415130'
+          required: true
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/file_creation_request'
+      responses:
+        '201':
+          description: Created. The metadata was saved sucessfully.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  file_uuid:
+                    type: string
+                    example: 'b96bdc16-8f27-44aa-9758-b4e5f13060fe'
+        '500':
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error_response'
+
+  /files/{user_uuid}/{file_uuid}:
+    get:
+      tags: [ 'File' ]
+      description: Get the metadata of the given file. This endpoint is suposed to only be used by the gateway service to obtain the location (files/volume/archive_uuid) of the file.
+      parameters:
+        - in: path
+          name: user_uuid
+          schema:
+            type: string
+            example: '658b4e63-b5ac-46a7-ac43-efb6a1415130'
+          required: true
+        - in: path
+          name: file_uuid
+          schema:
+            type: string
+            example: 'b96bdc16-8f27-44aa-9758-b4e5f13060fe'
+          required: true
+      responses:
+        '200':
+          description: Ok. The metadata was returned successfuly.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/metadata'
+        '403':
+          description: Forbidden. The file is not owned by the user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error_response'
+        '404':
+          description: Not found. No file with the given file_uuid was found.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error_response'
+        '500':
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error_response'
+    delete:
```

Comment:
As discussed with @shoriwe it'd be convenient to being able to delete multiple files at once, this is yet to be added to [Filesystem's use cases](https://github.com/hawks-atlanta/docs/blob/main/Use%20cases.md#filesystem).

Reply:
We´re not including this for now.